File: Mavim-apps/Mavim.frontend/projects/mavim-app/src/shared/field/mappers/multitextfield.mapper.ts
import { Field } from '../models/field.model';
import { MultiTextField } from '../models/api';
import { GenericTextFieldMapper } from './abstract/generic-text-field.mapper';
import { mapToApiFieldType, mapToFieldType } from './fieldvaluetype';

export class MultiTextFieldMapper extends GenericTextFieldMapper<MultiTextField> {
	public get getUrlTypeName(): string {
		return 'multitext';
	}

	public getMappedField(field: MultiTextField): Field {
		return this.mapField(field);
	}

	public getMappedApiField(field: Field): MultiTextField {
		return this.mapApiField(field);
	}

	protected mapApiField(field: Field): MultiTextField {
		const data = field.data ? [...field.data] : undefined;
		return {
			...field,
			fieldValueType: mapToApiFieldType(field.fieldValueType),
			data
		};
	}

	protected mapField(field: MultiTextField): Field {
		return {
			...field,
			fieldValueType: mapToFieldType(field.fieldValueType),
			options: undefined,
			data: field.data ? [...field.data] : []
		} as Field;
	}
}

File: Mavim-services/Topic/Mavim.Manager.Api.Topic.Repository/v1/Mappers/MultiTextFieldMapper.cs
using Mavim.Manager.Api.Topic.Repository.Interfaces.v1.enums;
using Mavim.Manager.Api.Topic.Repository.Interfaces.v1.Fields;
using Mavim.Manager.Api.Topic.Repository.v1.Fields;
using Mavim.Manager.Api.Topic.Repository.v1.Mappers.Abstract;
using System;
using System.Linq;

namespace Mavim.Manager.Api.Topic.Repository.v1.Mappers
{
    internal class MultiTextFieldMapper : GenericTextFieldMapper<Model.IMultiValueSimpleField, IMultiTextField>
    {
        /// <summary>
        /// Gets the generic mapped multi text service field.
        /// </summary>
        /// <param name="field">The field.</param>
        /// <returns></returns>
        protected override IMultiTextField GetGenericMappedServiceField(Model.IMultiValueSimpleField field)
        {
            if (field == null) throw new ArgumentNullException(nameof(field));

            return new MultiTextField
            {
                FieldId = field.FieldDefinition.ID,
                FieldName = field.FieldDefinition.Name,
                SetOrder = field.FieldSetDefinition.OrderNumber,
                Order = field.OrderNumber,
                FieldSetId = field.FieldSetDefinition.ID,
                SetName = field.FieldSetDefinition?.Name ?? string.Empty,
                FieldValues = field.FieldValues?.Select(i => i?.ToString()),
                FieldValueType = FieldType.MultiText,
                Required = field.FieldDefinition.Required,
                Readonly = field.ReadOnly
            };
        }

        /// <summary>
        /// Gets the generic mapped multi text repo field.
        /// </summary>
        /// <param name="field">The field.</param>
        /// <param name="simpleField">The simple field.</param>
        /// <returns></returns>
        /// <exception cref="ArgumentNullException">field</exception>
        protected override object[] GetGenericMappedRepoField(IMultiTextField field, Model.ISimpleField simpleField = null)
        {
            if (field == null) return new object[] { null };

            return field.FieldValues == null ? new object[] { null } : field.FieldValues.ToArray<object>();
        }
    }
}